packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -2024,23 +2024,55 @@ describe('ComboBox', function () {
       );
 
       let button = getByRole('button');
+      let combobox = getByRole('combobox');
 
-      await waitFor(() => expect(load).toHaveBeenCalledTimes(1));
+      expect(load).toHaveBeenCalledTimes(1);
       expect(onOpenChange).toHaveBeenCalledTimes(0);
       expect(onLoadMore).toHaveBeenCalledTimes(0);
 
+      // open menu
       await act(async () => {
         triggerPress(button);
-        jest.runAllTimers();
+        // advance to open state from Transition
+        jest.advanceTimersToNextTimer();
       });
-
       let listbox = getByRole('listbox');
       expect(listbox).toBeVisible();
+      jest.spyOn(listbox, 'clientHeight', 'get').mockImplementation(() => 100);
+      // update size, virtualizer raf kicks in
+      act(() => {jest.advanceTimersToNextTimer();});
+      // onLoadMore queued by previous timer, run it now
+      act(() => {jest.advanceTimersToNextTimer();});
 
       expect(onOpenChange).toHaveBeenCalledTimes(1);
       expect(onOpenChange).toHaveBeenCalledWith(true, 'manual');
       expect(onLoadMore).toHaveBeenCalledTimes(1);
-      await waitFor(() => expect(load).toHaveBeenCalledTimes(1));
+      expect(load).toHaveBeenCalledTimes(1);
+
+      if (React.version.startsWith('17') || React.version.startsWith('16')) {
```

Comment:
the order of these events and what they are tied to varies from 16/17 -> 18 and I don't understand why